### PR TITLE
Refactor project

### PR DESCRIPTION
**Pull request by ramesius**
- Fix up some promise chaining
- Modularize the code a bit more by breaking out into common functions
- Move functionality into separate files, split by concern (start/stop, asg/instances)

Sorry again for the large PR, got a bit carried away.
Will likely need some testing, perhaps running it in our own account.

### CODE REVIEW

**Comment by ramesius** [comment]
Also sets latest node js version for AWS lambda.

**Comment by krutisfood** [comment]
Looking at testing this internally now, will report back.

File: src/stop.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      }
+
+      stoppableASGs.forEach((asg) => {
+        console.log(asg.AutoScalingGroupName);
+      });
+      return tagASGs(stoppableASGs);
+    })
+    .then((taggedASGs) => {
+      console.log(`Finished tagging ASGs. Moving on to spin down ${taggedASGs.length} of them.`);
+      return stopASGs(taggedASGs);
```
Comment (krutisfood) [inline src/stop.js]: Getting an error
```TypeError: stopASGs is not a function    at listASGsToStop.then.then (/var/task/src/stop.js:24:14)    at process._tickDomainCallback (internal/process/next_tick.js:135:7) 2017-05-03T12:39:41.844Z    839682e8-2ffd-11e7-b4dd-a7ea8e9b680f    {"errorMessage":"stopASGs is not a function","errorType":"TypeError","stackTrace":["listASGsToStop.then.then (/var/task/src/stop.js:24:14)","process._tickDomainCallback (internal/process/next_tick.js:135:7)"]} END RequestId: 839682e8-2ffd-11e7-b4dd-a7ea8e9b680f```
Suspect it's `stopASGs.stopASGs(taggedASGs);`?

Comment (ramesius) [inline src/stop.js]: Ahh yes, the export for `stopASGs` looks like it is just meant to be the function instead of an object export, will fix this up.

**Comment by krutisfood** [comment]
Ok scheduled for run tonight, this time for sure!

**Pull request merged**